compiler/rustc_mir/src/borrow_check/type_check/mod.rs

```diff
@@ -1298,7 +1298,7 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                         };
                         debug!("opaque_defn_ty = {:?}", opaque_defn_ty);
                         let subst_opaque_defn_ty =
-                            opaque_defn_ty.concrete_type.subst(tcx, opaque_decl.substs);
+                            opaque_defn_ty.concrete_type.subst(tcx, opaque_decl.substs[0]);
```

Comment:
Should we have assertions that `opaque_decl.substs.len() == 1`?

Reply:
it doesn't need to be 1, we just ignore all others, since we already errored. Technically we could iterate, but I've tried this and in practice it doesn't give us more or more useful diagnostics.